platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java

```diff
@@ -189,7 +190,28 @@ public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(
                     "Component wiring does not support primitive input types. Use a boxed primitive instead.");
         }
 
-        return getOrBuildInputWire(proxy.getMostRecentlyInvokedMethod(), null, null, handler);
+        return getOrBuildInputWire(proxy.getMostRecentlyInvokedMethod(), null, null, handler, null);
+    }
+
+    /**
+     * Get an input wire for this component.
+     *
+     * @param handler      the component method that will handle the input, e.g. "MyComponent::handleInput". Should be a
+     *                     method on the class, not a method on a specific instance.
+     * @param <INPUT_TYPE> the input type
+     * @return the input wire
+     */
+    public <INPUT_TYPE> InputWire<INPUT_TYPE> getInputWire(@NonNull final Consumer<COMPONENT_TYPE> handler) {
+        Objects.requireNonNull(handler);
+
+        try {
+            handler.accept(proxyComponent);
+        } catch (final NullPointerException e) {
+            throw new IllegalStateException(
+                    "Component wiring does not support primitive input types. Use a boxed primitive instead.");
```

Comment:
Minor: might want to put the `e` into the `cause` of your new exception to provide more information when debugging issues. You can never really be sure (now, or in the future as the code grows) what exactly caused an NPE w/o seeing the original exception.

Reply:
Added